File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/call/LineCall.java
package com.flat.simplex.parser.logic.line.call;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;
import com.flat.simplex.parser.logic.Context;
import com.flat.simplex.parser.logic.line.LineValue;

public class LineCall {

    public enum Type {
        Field, IndexCall, MethodCall, Struct, Array, Function, Group, Value
    }

    private LineCall next;
    private LineCall prev;
    private final Type type;
    private final Token token;
    private final Block parent;

    public LineCall(Block parent, Token token, Type type) {
        this.parent = parent;
        this.token = token;
        this.type = type;
    }

    public void setNext(LineCall next) {
        this.next = next;
        next.setPrev(this);
    }

    public LineCall getNext() {
        return next;
    }

    public void setPrev(LineCall prev) {
        this.prev = prev;
    }

    public LineCall getPrev() {
        return prev;
    }

    public Type getType() {
        return type;
    }

    public Token getToken() {
        return token;
    }

    public Block getParent() {
        return parent;
    }

    public LineValue.ReturnType getReturnType() {
        return LineValue.ReturnType.Object;
    }

    public void load() {

    }
}

File: SimplexApp/simplex/lang/parser/logic/line/call/LineCall.js
var LineValue, Error;

class LineCall {
    constructor(parent, token, type) {
        this.next = null;
        this.prev = null;

        this.parent = parent;
        this.token = token;
        this.type = type;
    }

    setNext(next) {
        this.next = next;
        this.next.setPrev(this);
    }

    getNext() {
        return this.next;
    }

    setPrev(prev) {
        this.prev = prev;
    }

    getPrev() {
        return this.prev;
    }

    getType() {
        return this.type;
    }

    getToken() {
        return this.token;
    }

    getParent() {
        return this.parent;
    }

    getReturnType() {
        return LineValue.Object;
    }

    load() {

    }
}

LineCall.Field = 'Field';
LineCall.IndexCall = 'IndexCall';
LineCall.MethodCall = 'MethodCall';
LineCall.Struct = 'Struct';
LineCall.Array = 'Array';
LineCall.Function = 'Function';
LineCall.Group = 'Group';
LineCall.Value = 'Value';

module.exports = LineCall;
LineValue = require("simplex/lang/parser/logic/line/LineValue.js");
Error = require("simplex/lang/parser/logic/error/Error.js");